apps/mobile/src/libs/components/input/input.tsx

```diff
@@ -5,49 +5,116 @@ import {
 	type FieldPath,
 	type FieldValues,
 } from "react-hook-form";
-import { TextInput } from "react-native";
 
-import { Text, View } from "~/libs/components/components";
-import { useFormController } from "~/libs/hooks/hooks";
+import {
+	LinearGradient,
+	Text,
+	TextInput,
+	View,
+} from "~/libs/components/components";
+import { BaseColor, GradientColor } from "~/libs/enums/enums";
+import { useFormController, useState } from "~/libs/hooks/hooks";
+import { globalStyles } from "~/libs/styles/styles";
 
 import { styles } from "./styles";
 
 type Properties<T extends FieldValues> = {
+	accessoryRight?: JSX.Element;
 	control: Control<T, null>;
 	errors: FieldErrors<T>;
+	isAutoFocused?: boolean;
 	isSecureTextEntry?: boolean;
 	label: string;
 	name: FieldPath<T>;
 	placeholder: string;
 };
 
 const Input = <T extends FieldValues>({
+	accessoryRight,
 	control,
 	errors,
+	isAutoFocused = false,
 	isSecureTextEntry = false,
 	label,
 	name,
 	placeholder,
 }: Properties<T>): JSX.Element => {
+	const [isFocused, setIsFocused] = useState<boolean>(false);
+
 	const { field } = useFormController({ control, name });
 
-	const { onBlur, onChange, value } = field;
+	const { onChange, value } = field;
 
 	const error = errors[name]?.message;
 	const hasError = Boolean(error);
+	const FIRST_COLOR_STOP = -0.4;
+	const SECOND_COLOR_STOP = 0.9;
+	const GRADIENT_ANGLE = 305;
+	const ANGLE_CENTER_POINT = 0.5;
```

Comment:
Too many constants in this file, create a separate constants folder. Also I believe ANGLE_CENTER_POINT can be used in other places, if that's the case, move to common constants

Reply:
All this constants are used together two times: in GradientText and Button, where LinearGradient is used. And might be used if we need to create Icon with gradient MaskView. So I found that common enum could be better in this case, because constants are used together.
Can I make it this way?
```js
const AngleGradient = {
	ANGLE: 305,
	FIRST_STOP: -0.4,
	SECOND_STOP: 0.9,
	X_POINT: 0.5,
	Y_POINT: 0.5,
} as const;
```

Reply:
Yes, I like the idea of creating an enum